dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/TrackerUserService.java

```diff
@@ -0,0 +1,72 @@
+package org.hisp.dhis.tracker;
+
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.hibernate.HibernateUtils;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserCredentials;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+/**
+ * Specialized User Service for Tracker that executes User look-up in a read-only transaction
+ *
+ * @author Luciano Fiandesio
+ */
+@Service
+public class TrackerUserService
+{
+    private final CurrentUserService currentUserService;
+
+    private final IdentifiableObjectManager manager;
+
+    public TrackerUserService( CurrentUserService currentUserService, IdentifiableObjectManager manager )
+    {
+        this.currentUserService = currentUserService;
+        this.manager = manager;
+    }
+
+    /**
+     * Fetch a User by user uid
+     * @param userUid a User uid
+     * @return a User
+     */
+    @Transactional( readOnly = true )
+    public User getUser( String userUid )
```

Comment:
I'm wondering if this is really specific to tracker. Looks fairly generic and might be used everywhere. Could we move it to CurrentUserService or similar @luciano-fiandesio ?

Reply:
I agree, I was also unsure how to design this. I don't think that `CurrentUserService` is the right place, because this service requires a `uid` (so it'n not the "current user"). We have a `UserService` which has `getUser(String uid)` method, but it does not initialize the collections like I have to do here. I see this PR more like a temporary fix: we are looking at several design "issues" in the new Tracker with @stian-sandvold  and one of those is definitively the way Users are handled. Currently, without this PR, the new Tracker doesn't work **AT ALL**, so this is more a fix to allow the testing to proceed while we try to redesign some of the internals. Also, the whole user management is getting redesigned (User , UserCredentials) so I hope this collections initialization is no longer going to be needed.